### DISCUSSION TITLE
Expand Props in args table

### DISCUSSION THREAD

**Discussion by jdiponziano** [storybookjs/storybook#25201] [Help]
### Summary

I have a large complex component that has many proptypes for each of its props. In this image, how do i get the ProfileBadgeTypes to show whats in it like the ship prop and not just say ProfileBadgeTypes

![Screenshot 2023-12-12 at 10 50 14 AM](https://github.com/storybookjs/storybook/assets/12420501/90d6e01a-fa2f-4f3c-8fa0-a12538e1c54e)

### Additional information

```
import type { StorybookConfig } from "@storybook/react-vite";
import { join, dirname, resolve } from "path";
import { mergeConfig } from "vite";

/**
 * This function is used to resolve the absolute path of a package.
 * It is needed in projects that use Yarn PnP or are set up within a monorepo.
 */
function getAbsolutePath(value: string): string {
  return dirname(require.resolve(join(value, "package.json")));
}
const config: StorybookConfig = {
  stories: [
    "../src/**/*.mdx",
    "../src/**/*.stories.@(js|jsx|mjs|ts|tsx)",
    "../../../packages/**/*.mdx",
    "../../../packages/**/**/*.stories.@(ts|tsx)",
  ],
  addons: [
    getAbsolutePath("@storybook/addon-links"),
    getAbsolutePath("@storybook/addon-essentials"),
    getAbsolutePath("@storybook/addon-onboarding"),
    getAbsolutePath("@storybook/addon-interactions"),
    getAbsolutePath("@storybook/addon-a11y"),
    getAbsolutePath("@storybook/addon-jest"),
    getAbsolutePath("@storybook/addon-designs"),
  ],
  framework: {
    name: getAbsolutePath("@storybook/react-vite"),
    options: {},
  },
  docs: {
    autodocs: true,
  },
  typescript: {
    check: false,
    reactDocgen: "react-docgen-typescript",
    reactDocgenTypescriptOptions: {
      setDisplayName: false,
      shouldExtractLiteralValuesFromEnum: true,
      shouldRemoveUndefinedFromOptional: true,
      compilerOptions: {
        allowSyntheticDefaultImports: false,
        esModuleInterop: false,
        preserveSymlinks: true,
      },
      propFilter: (prop) => {
        if (prop.name === "children") {
          return true;
        }

        if (prop.parent) {
          return (
            !/@types\/react/.test(prop.parent.fileName) &&
            !/@emotion/.test(prop.parent.fileName) &&
            !/\/@types\/react-/.test(prop.parent.fileName)
          );
        }
        return true;
      },
      include: ["../../packages/ui/src/**/*"],
    },
  },
  ViteFinal: async (config) => {
    if (config.resolve) {
      return mergeConfig(config, {
        resolve: {
          alias: {
            ...config.resolve.alias,
            ui: resolve(__dirname, "../../../packages/ui/src"),
          },
        },
      });
    }
  },
};
export default config;
```

**Comment by dilipk3** [comment]
@jdiponziano I'm not a maintainer of Storybook but I too was looking into this and here are some of my thoughts.

This seems to be a limitation of the underlying static analyzing tool `react-docgen-typescript`. It doesn't unwrap type aliases. If you inline your type, it should work(I'm assuming thats why the ship example works)

Alternatively, you can can set `reactDocgen` to `react-docgen` which uses the another tool, that enables expanding types but only one level deep. But `react-docgen` may have other limitations w.r.t parsing typescript. 🤷‍♂️

**Reply by smlimasu** [reply]
@dilipk3 you could also leverage the "arg types Enhancers" property f.ex. in preview to put the content of the interface/type into a drop-down of argtypes controls as an overlay (see argtypes documentation). On the meta level, you could manually type the Type, cause it gets stripped at runtime, or spread the interface, maybe formatted with prettier. If you wanted to have the type content automatically, then I think you would need to additionally leverage a lib like ts-morph to get that info into the argtypes or the enhancer function.
